src/mlmc/sample_storage.py

```diff
@@ -223,30 +224,50 @@ def sample_pairs(self):
         levels_results = list(np.empty(len(np.max(self._results.keys()))))
```

Comment:
The level_ids should be in fact level indices forming a sequence starting from 0 so this seems to be a bit too complicated. May the self._results be simply a list?

We should not support retrieving all samples at once as that may hang the computation when it doesn't fit into memory. Alternatively, we may support a shortcut of setting the chunk size to N followed by `samples(n_samples=N)`.

Reply:
In general, we can obtain results from a certain level before results from for example zero level. So I think this implementation is reasonable. 

Retrieving all samples at once by default is used in `Memory()` storage. I would prefer the following approach: if the user thinks that some memory problems may occur, he should use another storage.